packages/synthetix-main/contracts/mixins/MarketManagerMixin.sol

```diff
@@ -143,7 +143,23 @@ contract MarketManagerMixin is MarketManagerStorage, PoolModuleStorage {
         return IMarket(marketData.marketAddress).reportedDebt(marketId);
     }
 
+    function _depositedCollateralValue(MarketData storage marketData) internal view returns (uint) {
+        uint totalDepositedCollateralValue = 0;
+
+        for (uint i = 0; i < marketData.depositedCollateral.length; i++) {
+            DepositedCollateral memory depositedCollateral = marketData.depositedCollateral[i];
+            totalDepositedCollateralValue +=
+                _getCollateralValue(depositedCollateral.collateralType) *
+                depositedCollateral.amount;
+        }
+
+        return totalDepositedCollateralValue;
+    }
+
     function _totalBalance(uint marketId, MarketData storage marketData) internal view returns (int) {
-        return int(IMarket(marketData.marketAddress).reportedDebt(marketId)) + marketData.issuance;
+        return
+            int(IMarket(marketData.marketAddress).reportedDebt(marketId)) +
+            marketData.issuance -
+            int(_depositedCollateralValue(marketData));
```

Comment:
maybe just pass in `marketData.depositedCollateral` to `_depositedCollateralValue` instead of the entire data.